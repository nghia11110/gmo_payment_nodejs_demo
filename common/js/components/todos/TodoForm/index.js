import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Form } from 'semantic-ui-react';
import classnames from 'classnames/bind';
import css from './index.scss';

class TodoForm extends Component {
  static propTypes = {
    onSubmit: PropTypes.func,
    className: PropTypes.string
  };

  state = { todoText: '' };

  submitTodo = ev => {
    ev.preventDefault();

    const { onSubmit } = this.props;
    const { todoText } = this.state;

    if (todoText && todoText !== '' && typeof onSubmit === 'function') {
      onSubmit(todoText);
      this.setState({ todoText: '' });
    }
  };

  onTodoChange = ev => {
    this.setState({ todoText: ev.target.value });
  };

  render() {
    const { className } = this.props;
    const { todoText } = this.state;

    return (
      <Form
        className={classnames(css.todoForm, className)}
        onSubmit={this.submitTodo}
      >
        <Form.Group>
          <Form.Input
            onChange={this.onTodoChange}
            value={todoText}
            type="text"
            placeholder="Add a todo..."
          />
          <Form.Button content="Add" icon="plus" />
        </Form.Group>
      </Form>
    );
  }
}

export default TodoForm;
