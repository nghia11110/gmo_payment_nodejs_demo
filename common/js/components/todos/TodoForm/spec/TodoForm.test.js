import React from 'react';
import TodoForm from '../index';
import { mount } from 'enzyme';

describe('TodoForm', () => {
  it('renders correctly', () => {
    const component = mount(<TodoForm />);
    expect(component).toMatchSnapshot();
  });

  describe('clicking on submit button', () => {
    test('calls the onSubmit prop', () => {
      const mockSubmit = jest.fn();
      const component = mount(<TodoForm onSubmit={mockSubmit} />);

      // test form submission
      component.setState({ todoText: 'Foobar' });
      component.find('form').simulate('submit');

      expect(mockSubmit.mock.calls.length).toEqual(1);
    });
  });
});
