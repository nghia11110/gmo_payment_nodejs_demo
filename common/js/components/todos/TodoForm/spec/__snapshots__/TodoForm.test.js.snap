// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm renders correctly 1`] = `
<TodoForm>
  <Form
    as="form"
    className="todoForm"
    onSubmit={[Function]}
  >
    <form
      className="ui form todoForm"
      onSubmit={[Function]}
    >
      <FormGroup>
        <div
          className="fields"
        >
          <FormInput
            as={[Function]}
            control={[Function]}
            onChange={[Function]}
            placeholder="Add a todo..."
            type="text"
            value=""
          >
            <FormField
              control={[Function]}
              onChange={[Function]}
              placeholder="Add a todo..."
              type="text"
              value=""
            >
              <div
                className="field"
              >
                <Input
                  onChange={[Function]}
                  placeholder="Add a todo..."
                  type="text"
                  value=""
                >
                  <div
                    className="ui input"
                  >
                    <input
                      key="text"
                      onChange={[Function]}
                      placeholder="Add a todo..."
                      type="text"
                      value=""
                    />
                  </div>
                </Input>
              </div>
            </FormField>
          </FormInput>
          <FormButton
            as={[Function]}
            content="Add"
            control={[Function]}
            icon="plus"
          >
            <FormField
              content="Add"
              control={[Function]}
              icon="plus"
            >
              <div
                className="field"
              >
                <Button
                  as="button"
                  content="Add"
                  icon="plus"
                >
                  <button
                    className="ui button"
                    onClick={[Function]}
                    role="button"
                  >
                    <Icon
                      as="i"
                      key="plus"
                      name="plus"
                    >
                      <i
                        aria-hidden="true"
                        className="plus icon"
                      />
                    </Icon>
                    Add
                  </button>
                </Button>
              </div>
            </FormField>
          </FormButton>
        </div>
      </FormGroup>
    </form>
  </Form>
</TodoForm>
`;
